sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/aggregator/DoubleHistogramAggregatorTest.java

```diff
@@ -0,0 +1,181 @@
+    aggregatorHandle.recordLong(20);
+    aggregatorHandle.recordLong(5);
+    aggregatorHandle.recordLong(150);
+    aggregatorHandle.recordLong(2000);
+    assertThat(aggregatorHandle.accumulateThenReset())
+        .isEqualTo(HistogramAccumulation.create(2175, new long[] {1, 1, 1, 1}));
+  }
+
+  @Test
+  void toAccumulationAndReset() {
+    AggregatorHandle<HistogramAccumulation> aggregatorHandle = aggregator.createHandle();
+    assertThat(aggregatorHandle.accumulateThenReset()).isNull();
+
+    aggregatorHandle.recordLong(100);
+    assertThat(aggregatorHandle.accumulateThenReset())
+        .isEqualTo(HistogramAccumulation.create(100, new long[] {0, 1, 0, 0}));
+    assertThat(aggregatorHandle.accumulateThenReset()).isNull();
+
+    aggregatorHandle.recordLong(0);
+    assertThat(aggregatorHandle.accumulateThenReset())
+        .isEqualTo(HistogramAccumulation.create(0, new long[] {1, 0, 0, 0}));
+    assertThat(aggregatorHandle.accumulateThenReset()).isNull();
+  }
+
+  @Test
+  void accumulateData() {
+    assertThat(aggregator.accumulateDouble(11.1))
+        .isEqualTo(HistogramAccumulation.create(11.1, new long[] {0, 1, 0, 0}));
+    assertThat(aggregator.accumulateLong(10))
+        .isEqualTo(HistogramAccumulation.create(10.0, new long[] {1, 0, 0, 0}));
+  }
+
+  @Test
+  void toMetricData() {
+    AggregatorHandle<HistogramAccumulation> aggregatorHandle = aggregator.createHandle();
+    aggregatorHandle.recordLong(10);
+
+    MetricData metricData =
+        aggregator.toMetricData(
+            Collections.singletonMap(Labels.empty(), aggregatorHandle.accumulateThenReset()),
+            0,
+            10,
+            100);
+    assertThat(metricData).isNotNull();
+    assertThat(metricData.getType()).isEqualTo(MetricDataType.HISTOGRAM);
+    assertThat(metricData.getDoubleHistogramData().getAggregationTemporality())
+        .isEqualTo(AggregationTemporality.DELTA);
+  }
+
+  @Test
+  void testHistogramCounts() {
+    assertThat(aggregator.accumulateDouble(1.1).getCounts().length)
+        .isEqualTo(boundaries.length + 1);
+    assertThat(aggregator.accumulateLong(1).getCounts().length).isEqualTo(boundaries.length + 1);
+
+    AggregatorHandle<HistogramAccumulation> aggregatorHandle = aggregator.createHandle();
+    aggregatorHandle.recordDouble(1.1);
+    HistogramAccumulation histogramAccumulation = aggregatorHandle.accumulateThenReset();
+    assertThat(histogramAccumulation).isNotNull();
+    assertThat(histogramAccumulation.getCounts().length).isEqualTo(boundaries.length + 1);
+  }
+
+  @Test
+  void testMultithreadedUpdates() throws Exception {
+    final AggregatorHandle<HistogramAccumulation> aggregatorHandle = aggregator.createHandle();
+    final Histogram summarizer = new Histogram();
+    int numberOfThreads = 10;
+    final long[] updates = new long[] {1, 2, 3, 5, 7, 11, 13, 17, 19, 23};
+    final int numberOfUpdates = 1000;
+    final CountDownLatch startingGun = new CountDownLatch(numberOfThreads);
```

Comment:
```suggestion
    final CountDownLatch starter = new CountDownLatch(numberOfThreads);
```

Let's avoid words like gun

Reply:
https://www.loc.gov/law/help/usconlaw/second-amendment.php#:~:text=The%20Second%20Amendment%2C%20one%20of,self%2Devident%2C%20and%20has%20given